### PR TITLE
Add a new interface for creating a channel in android tester app

### PR DESCRIPTION
**Pull request by zsurocking**

### CODE REVIEW

File: android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java [hunk 1]
```diff
@@ -124,39 +110,11 @@ public void onCompleted() {
 
 
   public InteropTester(String testCase,
-                       String host,
-                       int port,
-                       @Nullable String serverHostOverride,
-                       boolean useTls,
-                       @Nullable InputStream testCa,
-                       @Nullable String androidSocketFactoryTls,
+                       TesterChannelBuilder builder,
```
Comment (ejona86) [inline android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java]: Why not just pass in a ManagedChannel?

File: android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java [hunk 2]
```diff
@@ -124,39 +110,11 @@ public void onCompleted() {
 
 
   public InteropTester(String testCase,
```
Comment (ejona86) [inline android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java]: How does this PR compile? You didn't change any usages of this constructor.

Similarly, there are no usages of TesterOkHttpChannelBuilder.

File: android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java (hunk 1 above)
Comment (zsurocking) [inline android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java]: Done

File: android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java (hunk 2 above)
Comment (zsurocking) [inline android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java]: Yeah, android-integration-test does not automatically run. PTAL

File: android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterChannelBuilder.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+package io.grpc.android.integrationtest;
+
+import io.grpc.ManagedChannel;
+
+/**
+ * A interface to build a ManagedChannel for test use.
+ */
+public interface TesterChannelBuilder {
```
Comment (ejona86) [inline android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterChannelBuilder.java]: This is unused.

File: android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterOkHttpChannelBuilder.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.security.auth.x500.X500Principal;
+
+/**
+ * This implementation creates a OkHttp based channel.
+ */
+public class TesterOkHttpChannelBuilder implements TesterChannelBuilder{
```
Comment (ejona86) [inline android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterOkHttpChannelBuilder.java]: Feel free to make this just have a static helper method. Doesn't matter much to me.

**Comment by ejona86** [comment]
@zsurocking LGTM

**Comment by ejona86** [comment]
@zsurocking, don't forget to change the commit description before merging to master.

File: android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterOkHttpChannelBuilder.java (hunk 4 above)
Comment (zsurocking) [inline android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterOkHttpChannelBuilder.java]: Done

File: android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterChannelBuilder.java (hunk 3 above)
Comment (zsurocking) [inline android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterChannelBuilder.java]: Removed

**Pull request merged**